### ISSUE TITLE
[BUG][Python] Composed Model serialization bug (anyOf)

### ISSUE THREAD

**Issue by gbvanrenswoude** [OpenAPITools/openapi-generator#10719]
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

When generating a client from an OAS with a composed model situation (like https://gist.github.com/gbvanrenswoude/1143a79dd6f150931c5c00826f6454a2) the resulting EventModel class does contain wrong information about the possible Models being used:

openapi_client/model/event_model.py 
```
    def openapi_types():
        """
        This must be a method because a model may have properties that are
        of type self, this must run after the class is loaded

        Returns
            openapi_types (dict): The key is attribute name
                and the value is attribute type.
        """
        return {
            'event_payload': (bool, date, datetime, dict, float, int, list, str, none_type,),  # noqa: E501
            'correlatie_id': (str,),  # noqa: E501
            'event_title': (str,),  # noqa: E501
        }
```

Where 

```py
'event_payload': (bool, date, datetime, dict, float, int, list, str, none_type,),  # noqa: E501
```

is wrong. This might indicate a composed model serialization bug caused by the use of anyOf in the OAS.

##### replicate

In order to replicate the situation, save the linked gist as whoof.json.
The command then used is `openapi-generator generate -i whoof.json -g python -o resources/whoof_api_client`

##### openapi-generator version

5.3.0

With 5.2.0 the issue exists too, but results in `'event_payload': (dict,),  # noqa: E501`.

##### related issues 
This might be related to this PR: 
- https://github.com/OpenAPITools/openapi-generator/pull/10334

**Comment by gbvanrenswoude** [comment]
We succesfully implemented the example being added in #10754. I will close this issue with that. That being said, the less pressing question for the future would be to support model composition in the future in properties. I will leave the example up and add the changed example below for any folks running into the same / similar issues with model composition.

```json
"components": {
    "schemas": {
      "fixed": {
        "anyOf": [
          {
            "$ref": "#/components/schemas/aaa"
          },
          {
            "$ref": "#/components/schemas/bbb"
          },
          {
            "$ref": "#/components/schemas/ccc"
          },
          {
            "$ref": "#/components/schemas/ddd"
          }
        ]
      },
      "EventModel": {
        "title": "Event Model",
        "required": [
          "aaa",
          "bbb",
          "ccc",
          "event_payload",
        ],
        "type": "object",
        "properties": {
          "event_payload": {
            "$ref": "#/components/schemas/fixed"
          }, 
....
```

**Comment by spacether** [comment]
I am working on a new python-experimental generator which will support composition in all Schema locations:
- models (CodegenModel)
- properties/additionalProperties/items (CodegenProperty)
- payload responses (CodegenResponse)
- endpoint parameters (CodegenParameter)
